### ISSUE TITLE
Cannot get Class for custom node.

### ISSUE THREAD

**Issue by Parallel-Paradox** [godotengine/godot#69534]
### Godot version

v4.0.beta7.mono.official [0bb1e89fb]

### System information

Windows 11

### Issue description

I create a addon in Godot4.0 by C#, and add some custom node by using AddCustomType.
When I press "create new node" in hierarchy, I got "Cannot get Class" error.
It seems doen't bother execution of program, but still annoying. 
![image](https://user-images.githubusercontent.com/81791934/205439559-eb7f9a91-bbce-4bd2-93e2-46e8167a39a6.png)

### Steps to reproduce

Create a custom node according to https://docs.godotengine.org/en/latest/tutorials/plugins/editor/making_plugins.html.

Then click "Creat New Node" in editor or search the name of custom node in the popup windows.

### Minimal reproduction project

https://github.com/Parallel-Paradox/ScatterCS.git

**Comment by Parallel-Paradox** [comment]
Reproduce this with gdscript.

**Comment by adamscott** [comment]
@Parallel-Paradox Could you add a minimal reproduction project in GDScript (in your description)? Doesn't have to be a git repo, it can be a .zip file (preferred)

**Comment by Parallel-Paradox** [comment]
@adamscott Here.

[CustomNode.zip](https://github.com/godotengine/godot/files/10147801/CustomNode.zip)

<img width="855" alt="1670124304289" src="https://user-images.githubusercontent.com/81791934/205472760-35491d63-8291-4097-a259-1538e6ce88b1.png">

Besides, I've tried this on macOS and this error didn't occur. And I roll back to Godot-4-beta6, -beta5, but this error still pop on windows.

It might not be a general problem?

**Comment by Parallel-Paradox** [comment]
I fix this error by adding these line to the custom_node.gd:

```gdscript
class_name CustomNode
@icon("res://addons/node/custom_node.svg")

@export_category("CustomNode")
@export var number = 5;
```

**Comment by Parallel-Paradox** [comment]
Since there's no "class_name" in C# as far as I know, I can't try this solution on the mono version of the same issue.

**Comment by bsil78** [comment]
Seems related to #61396

**Comment by Parallel-Paradox** [comment]
> Seems related to #61396 and #21789

@bsil78 I did some tests according the these issues.

I have registered this node as "CustomNode" in addon register script:

```gdscript
# plugin.gd
@tool
extends EditorPlugin

func _enter_tree():
	# Initialization of the plugin goes here.
	add_custom_type("CustomNode", "Node3D", preload("custom_node.gd"), preload("custom_node.svg"));
```

And all code below will print "Node3D":

```gdscript
# custom_node.gd
@tool
extends Node3D

func _ready():
	print(get_class())
```

```gdscript
# custom_node.gd
@tool
extends Node3D

func _ready():
	print(self.get_class())
```

```gdscript
# custom_node.gd
@tool
class_name CustomNode
extends Node3D

func _ready():
	print(get_class())
```

```gdscript
# custom_node.gd
@tool
class_name CustomNode
extends Node3D

func _ready():
	print(self.get_class())
```

**Comment by Haatchoum** [comment]
To add to this issue, I've been messing around with beta 8 Mono yesterday. 
I couldn't manage to create a custom node/resource hierarchy using the given plugin setup example in the docs. 
I have the same error as given by the author, even with "empty/wrapper" class. And the tree doesn't show any child of the initial custom node/resource. 

I wish the C# custom node/resource was available by using c# class attribute like this: 
`
[AddCustomType(from(base_class)]
public class my_new_class : base_class
{
//your class code here
}
`

**Comment by adamscott** [comment]
Relates to #69889

**Comment by Parallel-Paradox** [comment]
> Relates to #69889

@adamscott @Haatchoum  Yes, when I find `class_name` feature of gdscript, I noticed that there's an overlap between this and `add_custom_type`.

The only reason that I still reserve this issue is that I can't find a corresponding solution in C# and GDExtension.

**Comment by adamscott** [comment]
@Parallel-Paradox Can you create a MRP in C#? @raulsntos needs one.

**Comment by raulsntos** [comment]
Isn't this a Core issue though? If it's reproducible in GDScript. Also, isn't this issue a duplicate of https://github.com/godotengine/godot/issues/69889 then?

**Comment by adamscott** [comment]
> Isn't this a Core issue though? If it's reproducible in GDScript.

Yes, you're right. It happens in C# because of the use of `add_custom_type()`

> Also, isn't this issue a duplicate of #69889 then?

This issue predates #69889, so #69889 is the duplicate.

**Comment by Parallel-Paradox** [comment]
@adamscott @raulsntos Here is C# MRP.

Click this, it will print the error.
<img width="1352" alt="image" src="https://user-images.githubusercontent.com/81791934/207372703-e21004c6-ec74-426b-ad0e-bd3c57f22b46.png">
[MRP.zip](https://github.com/godotengine/godot/files/10219386/MRP.zip)

**Comment by Parallel-Paradox** [comment]
https://github.com/m50/Godot-CSharp-Node-Exports

Something like this might help..

**Comment by adamscott** [comment]
Just tested it. Fixed on master (c7c561f) by #69889.

CC @akien-mga

**Comment by Haatchoum** [comment]
Quick question as Beta 9 just came out. 
I reproduced a sample plugin project. I can confirm the base error doesn't reproduce anymore. So i suspect this fix has been added to the current Beta ?

Although, the underlying issue that prevents from creating a whole custom node hierarchy from plugin still exists as any node uploaded as child of another custom node isn't accessible. Do I miss something ?
